インクリメンタルサーチの実装

shukan0728 opened the pull request:
WHAT
インクリメンタルサーチの実装
WHY
チャット機能改善のため

shukan0728 committed:
インクリメンタルサーチ実装

shukan0728 committed:
チャットメンバーの追加と削除

yamamotokeishiro commented on app/assets/javascripts/user_inc.js:
```diff
@@ -0,0 +1,66 @@
+    $(`#chat-group-user-${id}`).remove();
+  })
+
+});
```
無駄な空白は削除しましょう！

yamamotokeishiro commented on app/assets/javascripts/user_inc.js:
```diff
@@ -0,0 +1,66 @@
+  $(function() {
+    $("#user-search-field").on("keyup", function() {
+      var input = $("#user-search-field").val();
+      $.ajax({
```
空白の文字が入力された際に、全件出てきてしまうかと思われますので対策しましょう！
例）
何も入力していない状態でdeleteキーを押すなど

yamamotokeishiro commented:
全体的にインデント、改行のミスが目立ちますので、全ファイル見直して修正しましょう！
見栄えが悪くなり勿体ないです

kohei-isoda commented on app/views/groups/_form.html.haml:
```diff
@@ -10,32 +10,24 @@
   .chat-group-form__field.clearfix
     .chat-group-form__field--left
-      %label.chat-group-form__label{:for => "chat_group_チャットメンバー"} チャットメンバー
+      %label.chat-group-form__label{for: "chat_group_チャットメンバー"} チャットメンバー
```
ヘルパーメソッドに変更いたしましょう！

kohei-isoda commented on app/controllers/users_controller.rb:
```diff
@@ -12,8 +12,15 @@ def update
+ end
 
+ def index
+  @users = User.where('name LIKE(?)', "%#{params[:name]}%")
```
current_userを@usersに入れないように記述いたしましょう！

kohei-isoda reviewed (commented):
修正お願い致します！

shukan0728 committed:
＃１修正/インクリメンタルサーチの実装

・ファイル全体を見て不必要な空行の消去
・_form.html.hamlビュー中のラベルをヘルパーメソッドに変更
・インクリサーチ中、current_userを含まないようにcontrollerのwhereメソッドを修正
・インクリサーチ中、名前が追加されていない場合はajaxが作動しないようif文を追加

MrUrakou commented on app/assets/javascripts/user_inc.js:
```diff
@@ -0,0 +1,67 @@
+    });
+  });
+
+    $(document).on('click','.user-search-add', function() {
```
documentから変更できますので、もう少し粘ってみてください。
あとインデント修正しましょう。
以下同様です。

MrUrakou commented on app/assets/javascripts/user_inc.js:
```diff
@@ -0,0 +1,67 @@
+  })
+
+  $(document).on('click', '.user-search-remove', function() {
+    var id = $(this).data('user-id');
```
この変数って使用されてますか？

MrUrakou reviewed (commented):
実際の挙動を確認したいので、キャプチャをプルリクエストに貼っていただけますか。

shukan0728 committed:
#2修正/インクリメンタルサーチの実装

・decument指定からクラス指定に変更
・インデント修正

shukan0728 committed:
#3修正/インクリメンタルサーチの実装

shukan0728 committed:
#3修正/インクリメンタルサーチの実装
